### PR TITLE
fix(ripple-effect): ripple displays on click or touch

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

The ripple effect is broken and does not display when clicking or tapping a button.

Issue URL: #25094

## What is the new behavior?

Reverts the optional chaining fixes for eslint and restores the original behavior for ripple effect.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Dev build: `6.0.17-dev.11649730310.17726968`

### CODE REVIEW

File: core/src/utils/tap-click.ts [hunk 1]
```diff
@@ -169,7 +170,8 @@ const getActivatableTarget = (ev: any): any => {
     const path = ev.composedPath() as HTMLElement[];
     for (let i = 0; i < path.length - 2; i++) {
       const el = path[i];
-      if (el?.classList.contains('ion-activatable')) {
+      // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
+      if (el.classList && el.classList.contains('ion-activatable')) {
```
Comment (liamdebeasi) [inline core/src/utils/tap-click.ts:174]: We don't need the optional check here anymore: https://developer.mozilla.org/en-US/docs/Web/API/Element/classList#browser_compatibility

The optional check was for IE support. All browsers that Ionic supports also support classList,

Comment (sean-perkins) [reply core/src/utils/tap-click.ts]: Discovered why this check was added. The composed path includes shadow dom elements, which don't have `classList` on their prototype. Likely `instance of ShadowRoot` did not exist back when that check was added.

Updated the logic to account for this.

Comment (liamdebeasi) [reply core/src/utils/tap-click.ts]: Great catch!

File: core/src/components/ripple-effect/test/basic/ripple-effect.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  const boundingBox = await el.boundingBox();
+
+  if (boundingBox) {
+    await page.mouse.move(boundingBox.x + boundingBox.width / 2, boundingBox.y + boundingBox.height / 2);
+    await page.mouse.down();
+  }
+
+  // Waits for the ripple effect to be added
+  await page.waitForSelector(`${selector}.ion-activated`);
```
Comment (amandaesmith3) [inline core/src/components/ripple-effect/test/basic/ripple-effect.e2e.ts:58]: If I check the button demo on https://ionicframework.com/docs, the button receives `ion-activated` for a brief moment on click, even with the busted ripple effect. Does the test fail with the broken state? 🤔

Comment (sean-perkins) [reply core/src/components/ripple-effect/test/basic/ripple-effect.e2e.ts:58]: Are you possibly being served from a cache? I don't see the class added to the button element when clicking and holding on the button in the docs examples. Chrome also doesn't indicate that the `class` attribute was modified during mouse down.

Comment (sean-perkins) [reply core/src/components/ripple-effect/test/basic/ripple-effect.e2e.ts:58]: Strange that doesn't happen on Mac 🤔  https://user-images.githubusercontent.com/13732623/163049314-6e795818-897e-464b-ba0c-ba589f0bd4fe.mp4

Anyways, I confirmed that with the changes on `main`, the test suite fails. With the changes in this branch, the test suite passes.

Comment (amandaesmith3) [reply core/src/components/ripple-effect/test/basic/ripple-effect.e2e.ts:58]: Cool, that's what's important 👍

**Review by amandaesmith3 — APPROVED** [review]

**Review by liamdebeasi — APPROVED** [review]
Great job!

**Pull request merged**